Functions/Quantification/xASL_quant_ASL.m

```diff
@@ -105,10 +105,17 @@
 PWI4D = double(PWI4D);
 M0_im = double(M0_im);
 
+% Create path to PWI3D based on the PWI4D path (Because PWI4D_Path can be a path to a merged NII)
+PWI3D_Path = PWI4D_Path;
```

Comment:
I would first take the file and change it from there, which seems a bit more secure to me:

[fPath, fFile] = xASL_fileparts(PWI4D_Path);
if ~strcmp(fFile(1:5), 'PWI4D')
      error('Incorrect PWI4D NIfTI path');
elseif length(fFile)==5
      fFile =[];
else
      fFile = fFile(6:end);
end
PWI3D_Path = fulllfile(fPath, ['PWI3D' fFile '.nii']);

Reply:
Yes. Sure, it's slightly more secure, so doing that. You can resolve.

Reply:
But it can be simplified and will work
[fPath, fFile, fExt] = xASL_fileparts(PWI4D_Path);
if ~strcmp(fFile(1:5), 'PWI4D')
	error('Incorrect PWI4D NIfTI path');
else
	PWI3D_Path = fulllfile(fPath, ['PWI3D' fFile(6:end) fExt]);
end

It will work also for PWI4D.nii.